addons/project/models/project.py

```diff
@@ -1795,6 +1795,13 @@ def write(self, vals):
             # reset kanban state when changing stage
             if 'kanban_state' not in vals:
                 vals['kanban_state'] = 'normal'
+        task_ids_without_user_set = set()
+        if 'user_ids' in vals and 'date_assign' not in vals:
+            # prepare update of date_assign after super call
+            for task in self:
+                if not task.user_ids:
+                    task_ids_without_user_set.add(task.id)
```

Comment:
You can do it in one line with a set comprehension. 🙂 
```suggestion
            task_ids_without_user_set = {task.id for task in self if not task.user_ids}
```
🙂